src/cdk/tree/tree.ts

```diff
@@ -371,4 +382,20 @@ export class CdkTreeNode<T> implements FocusableOption, OnDestroy {
     }
     this.role = 'treeitem';
   }
+
+  private _getParentNodeAriaLevel(): number {
+    let parent = this._elementRef.nativeElement.parentElement;
+    while (parent &&
+    !(parent.classList.contains('cdk-nested-tree-node') || parent.classList.contains('cdk-tree'))) {
+      parent = parent.parentElement;
+    }
+    if (!parent) {
+      throw Error('Incorrect tree structure containing detached node.');
+    } else if (parent.classList.contains('cdk-nested-tree-node')) {
+      return parseInt(parent.getAttribute('aria-level')!);
+    } else {
+      // parent.classList.contains('cdk-tree')
+      return 0;
+    }
+  }
```

Comment:
```suggestion
  function getParentNodeAriaLevel(nodeElement: HTMLElement): number {
    let parent = nodeElement;
    while (isNodeElement(parent)) {
      parent = parent.parentElement;
    }
    if (!parent) {
      if (isDevMode()) {
        throw Error('Incorrect tree structure containing detached node.');
      }
    } else if (parent.classList.contains('cdk-nested-tree-node')) {
      return Number(parent.getAttribute('aria-level')!);
    } else {
      // The ancestor element is the cdk-tree itself
      return 0;
    }
  }
  
  function isNodeElement(element: HTMLElement) {
    const classList = element?.classList;
    return !!(classList?.contains('cdk-nested-tree-node') || classList?.contains('cdk-tree'));
  }
```
* Moves from `private` member to module-level function (outside the class)
* Adds separate function to simplify the `while` condition
* Uses `Number` instead of `parseInt`
* Uses `isDevMode` to gate throwing the error
* Tweak comment to be more sentence-y